x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -509,31 +478,17 @@ public synchronized boolean isAuthorizationEngineAllowed() {
      *
      * @return {@code true} as long as the license is valid. Otherwise {@code false}.
      */
-    public synchronized boolean isWatcherAllowed() {
-        Status localStatus = status;
-
-        if (localStatus.active == false) {
-            return false;
-        }
-
-        switch (localStatus.mode) {
-            case TRIAL:
-            case GOLD:
-            case PLATINUM:
-            case STANDARD:
-                return true;
-            default:
-                return false;
-        }
+    public boolean isWatcherAllowed() {
+        return isAllowedByLicenseAndSecurity(OperationMode.STANDARD, false, true, true);
```

Comment:
I don't like the version of this method that passes `false` for security.
It's calling a method named isAllowedBy___Security` and then says "oh, but not really security".

Reading this method, on its own, if you aren't intimately familiar with what `isAllowedByLicenseAndSecurity` does, and what those parameters mean, a reader is left:

1. Wondering why watcher cares about security (because it's in that method name).
2. What all those parameters mean, and whether they're correct.

People adding a new license check for paid functionality shouldn't be expected to make decisions about so many parameters. There should be one obvious method with the smallest set of parameters that they can use.